src/docs/markdown/caddyfile/concepts.md

```diff
@@ -319,15 +320,16 @@ If your configuration relies on environment variables, you can use them in the C
 {$SITE_ADDRESS}
 ```
 
-Environment variables in this form are substituted before parsing begins, so they can expand to empty values, partial tokens, complete tokens, or even multiple tokens and lines.
+Environment variables in this form are substituted before Caddyfile parsing begins, so they can expand to empty values, partial tokens, complete tokens, or even multiple tokens and lines.
 
 A default value can be specified for when the environment variable is not found, by using `:` as the delimiter between the variable name and the default value:
 
 ```caddy
 {$DOMAIN:localhost}
 ```
 
-If you want to defer the substitution of an environment variable until runtime, you can use the [standard `{env.*}` placeholders](/docs/conventions#placeholders).
+If you want to defer the substitution of an environment variable until runtime, you can use the [standard `{env.*}` placeholders](/docs/conventions#placeholders). Note that not all parts of the config support these placeholders though, since a line of code is necessary to perform the replacement; placeholder replacement is not automatic and needs to be performed at runtime. If it doesn't seem to work, please file an issue to request support for it.
```

Comment:
> since a line of code is necessary to perform the replacement

I know we talked about this earlier in Slack, and I don't remember the full conversation now, but my reading of it tonight makes me think some users will assume this means they need to write a line of code to make placeholders work. I don't want to give that impression, so maybe this instead?

```suggestion
If you want to defer the substitution of an environment variable until runtime, you can use the [standard `{env.*}` placeholders](/docs/conventions#placeholders). Note that not all config parameters support placeholders yet, but most common ones do. If it doesn't seem to work for you, please file an issue to request support for it.
```

Reply:
Yeah but I _really_ want to imply somehow that "plugin/module authors need to implement it, otherwise it won't work". It's been a repeated thing where a plugin author didn't add support and we get questions about it on the forums.

Reply:
Made some adjustments here, hopefully it's better.